integration_tests/src/main/python/sort_test.py

```diff
@@ -224,21 +224,46 @@ def test_multi_orderby_with_limit_single_part(data_gen):
     assert_gpu_and_cpu_are_equal_collect(
             lambda spark : binary_op_df(spark, data_gen).coalesce(1).orderBy(f.col('a'), f.col('b').desc()).limit(100))
 
+
+@disable_ansi_mode  # ANSI mode is tested separately, in test_orderby_with_ansi_exceptions
```

Comment:
I don't understand why we would have a disable_ansi_mode.  IMO we should be explicitly setting ANSI mode the way the test needs it to be to test what we're trying to test, rather than skipping test coverage on platforms that happen to have it enabled by default.

Reply:
For the record, the aim of having `disable_ansi_mode` is to mark off the tests that might fail, for now.

This is only for getting the Spark 4 tests to pass.  We will have a separate pass to examine these, and possibly replace them with separate tests for ANSI on/off, or address the plugin's shortcomings in ANSI mode.

Reply:
I have refactored this now.  This tests with ANSI on/off, and with overflow exceptions.